homeassistant/components/python_script/__init__.py

```diff
@@ -216,16 +226,33 @@ def protected_getattr(obj, name, default=None):
         "hass": hass,
         "data": data or {},
         "logger": logger,
+        "output": {},
     }
 
     try:
         _LOGGER.info("Executing %s: %s", filename, data)
         # pylint: disable-next=exec-used
         exec(compiled.code, restricted_globals)  # noqa: S102
+        _LOGGER.debug(
+            "Output of python_script: `%s`:\n%s",
+            filename,
+            restricted_globals["output"],
+        )
+        # Ensure that we're always returning a dictionary
+        if not isinstance(restricted_globals["output"], dict):
+            output_type = type(restricted_globals["output"])
+            restricted_globals["output"] = {}
+            raise ScriptError(
+                f"Expected `output` to be a dictionary, was {output_type}"
+            )
     except ScriptError as err:
-        logger.error("Error executing script: %s", err)
+        raise ValueError(f"Error executing script: {err}") from err
```

Comment:
Note that bad user input during a service call should raise a `ServiceValidationError` nowadays.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

Reply:
Adjusted